pkg/serverless/trace/propagation/carriers.go

```diff
@@ -201,6 +213,51 @@ func snsEntityCarrier(event events.SNSEntity) (tracer.TextMapReader, error) {
 	return carrier, nil
 }
 
+// handleEventBridgeThroughSNS builds payload carrier for an EventBridge
+// event wrapped in an SNS message.
+func handleEventBridgeThroughSNS(eventBridgeEvent map[string]interface{}) (tracer.TextMapReader, error) {
+	detail, ok := eventBridgeEvent["detail"].(map[string]interface{})
+	if !ok {
+		return nil, errorUnsupportedPayloadType
+	}
+
+	datadogInfo, ok := detail["_datadog"].(map[string]interface{})
+	if !ok {
+		return nil, errorNoDDContextFound
+	}
+
+	carrier := make(tracer.TextMapCarrier)
+	for key, value := range datadogInfo {
+		if strValue, ok := value.(string); ok {
+			carrier[key] = strValue
+		}
+	}
+	return carrier, nil
+}
+
+// eventBridgeCarrier returns the tracer.TextMapReader used to extract trace
+// context from the events.EventBridgeEvent type.
+func eventBridgeCarrier(event events.EventBridgeEvent) (tracer.TextMapReader, error) {
+	// Check if trace context is in the Detail field
+	traceContext, ok := event.Detail[datadogTraceHeader].(map[string]interface{})
+	if !ok {
+		return nil, errorNoDDContextFound
+	}
+
+	carrier := make(tracer.TextMapCarrier)
+	for key, value := range traceContext {
+		if strValue, ok := value.(string); ok {
+			carrier[key] = strValue
+		}
+	}
+
+	if carrier["x-datadog-trace-id"] == "" || carrier["x-datadog-parent-id"] == "" {
+		return nil, errorNoDDContextFound
+	}
```

Comment:
Why are you checking on this specific keys?

Reply:
Yeah this is kind of unnecessary. As long as we check that `_datadog` exists, we should be good